blockchain-agent-proof/src/main/java/com/sphereon/alfresco/blockchain/agent/proof/BlockchainProofConfigVerifier.java

```diff
@@ -1,78 +1,84 @@
     private static final String ASSERT_MESSAGE_GET_CONFIG = "An error occurred whilst retrieving blockchain configuration %s. The response was empty.";
     private static final String EXCEPTION_MESSAGE_CREATE_CONFIG = "An error occurred whilst creating blockchain configuration %s: %s";
     private static final String EXCEPTION_MESSAGE_CREATE_CONFIG_API = "An error occurred whilst creating blockchain configuration %s, HTTP status: %d\n Response body:\n%s";
 
-    @Autowired
-    private ConfigurationApi configurationApi;
-
-    @Value("${blockchain.config-name:#{null}}")
-    protected String configName;
-
-    @Value("${blockchain.context:factom}")
-    protected String context;
+    private final ConfigurationApi configurationApi;
+    private final String configName;
+    private final String context;
 
+    public BlockchainProofConfigVerifier(final ConfigurationApi configurationApi,
+                                         @Value("${sphereon.blockchain.agent.blockchain-proof.config-name}") final String configName,
+                                         @Value("${sphereon.blockchain.agent.blockchain-proof.context:factom}") final String context) {
+        this.configurationApi = configurationApi;
+        this.configName = configName;
+        this.context = context;
+    }
 
     @Override
-    public Void execute() {
+    public void onApplicationEvent(@Nonnull final ApplicationReadyEvent applicationReadyEvent) {
+        verifyValidBlockchainProofApiConfig();
+    }
+
+    public void verifyValidBlockchainProofApiConfig() {
         try {
             final var configResponse = configurationApi.getConfiguration(configName);
             Assert.notNull(configResponse, String.format(ASSERT_MESSAGE_GET_CONFIG, configName));
             Assert.notNull(configResponse.getConfiguration(), String.format(ASSERT_MESSAGE_GET_CONFIG, configName));
-        } catch (ApiException e) {
-            if (e.getCode() == HttpStatus.NOT_FOUND.value()) {
-                createConfiguration();
-            } else {
+        } catch (final ApiException e) {
+            if (e.getCode() != HttpStatus.NOT_FOUND.value()) {
                 throw new RuntimeException(String.format(EXCEPTION_MESSAGE_GET_CONFIG, configName, "" + e.getCode()), e);
             }
-
-        } catch (Throwable throwable) {
-            throw new RuntimeException(String.format(EXCEPTION_MESSAGE_GET_CONFIG, configName, throwable.getMessage()), throwable);
+            createConfiguration();
+        } catch (final Exception exception) {
+            throw new RuntimeException(String.format(EXCEPTION_MESSAGE_GET_CONFIG, configName, exception.getMessage()), exception);
         }
-        return null;
     }
 
-
     private void createConfiguration() {
         try {
+            final var proofTypes = List.of(ContentRegistrationChainTypesEnum.PER_HASH_PROOF_CHAIN, ContentRegistrationChainTypesEnum.SINGLE_PROOF_CHAIN);
```

Comment:
This is all highle blockchain-proof specific and should be configurable in the future

Reply:
It is already in the proof-specific module because of that. So can be made configurable with back-wards compatible defaults.